src/mscorlib/src/System/TimeSpan.cs

```diff
@@ -405,6 +405,42 @@ public String ToString(String format, IFormatProvider formatProvider)
             return t1.Add(t2);
         }
 
+        public static TimeSpan operator *(TimeSpan timeSpan, double factor)
+        {
+            if (double.IsNaN(factor))
+            {
+                throw new ArgumentException(Environment.GetResourceString("Arg_CannotBeNaN"), nameof(factor));
+            }
+
+            double ticks = timeSpan.Ticks * factor;
+            if (ticks > long.MaxValue | ticks < long.MinValue)
+            {
+                throw new OverflowException(Environment.GetResourceString("Overflow_TimeSpanTooLong"));
+            }
+
+            return FromTicks((long)ticks);
+        }
+
+        public static TimeSpan operator *(double factor, TimeSpan timeSpan) => timeSpan * factor;
+
+        public static TimeSpan operator /(TimeSpan timeSpan, double divisor)
+        {
+            double ticks = timeSpan.Ticks / divisor;
+            if (ticks > long.MaxValue || ticks < long.MinValue || double.IsNaN(ticks))
+            {
+                // Assuming exceptions are the rarer case, and since both IsNaN(divisor) and 0/0 can
+                // result in IsNaN(ticks), defer determining which exception until this branch has
+                // been taken.
+                throw double.IsNaN(divisor)
```

Comment:
>double.IsNaN(divisor) [](start = 22, length = 21)

it is better to do this check first in the beginning of the method and then make the code cleaner. I understand you want to optimize but I think one check here is not really a big deal.